### PR TITLE
Add jira events module

### PR DESCRIPTION
**Pull request by oksanagorbachenko**

### CODE REVIEW

File: functions/report_push/jira_events.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from jira import JIRA
+
+API_URL = os.getenv("JIRA_URL")
+SECRET_NAME = os.getenv("SECRET_NAME", "ipa-develop-api-keys")
+REGION_NAME = os.getenv("REGION_NAME")
+SECRETS = get_secrets(REGION_NAME, SECRET_NAME)
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:6]: didn't get what the SECRETS and get_secrets

File: functions/report_push/jira_events.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+API_USERNAME = SECRETS.get("JIRA_API_USERNAME")
+API_PASSWORD = SECRETS.get("JIRA_API_PASSWORD")
+
+options = {'server': API_URL}
+jira = JIRA(options, basic_auth=(API_USERNAME, API_PASSWORD))
+
+
+def open_bug(project_key: str, table_name: str, fail_step: str, description: str, replaced_allure_links):
+    summary = "[DataQA][BUG][" + table_name + "] " + fail_step
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:15]: please use String Formatting

File: functions/report_push/push_data_report.py [hunk 3]
```diff
@@ -30,8 +34,20 @@ def handler(event, context):
     key = report.get('folder_key')
     run_name = report.get('run_name')
     items = []
-
+    failed_test = 0
     df = wr.s3.read_json(path=['s3://'+qa_bucket+'/allure/' + suite + '/' + key + '/allure-report/history/history-trend.json'])
+    result_df = wr.s3.read_json(path=['s3://'+qa_bucket+'/allure/' + suite + '/' + key + '/allure-report/results/'])
```
Comment (bvolodarskiy) [inline functions/report_push/push_data_report.py:39]: please use String Formatting
also 
s3.read_json have capability to filter on path_suffix when you're reading files

File: functions/report_push/push_data_report.py [hunk 4]
```diff
@@ -30,8 +34,20 @@ def handler(event, context):
...
+    for file_name in [file for file in os.listdir(result_df) if file.endswith('-result.json')]:
+        with open("result_df" + file_name) as json_file:
```
Comment (bvolodarskiy) [inline functions/report_push/push_data_report.py:41]: please use String Formatting

File: functions/report_push/push_data_report.py [hunk 5]
```diff
@@ -30,8 +34,20 @@ def handler(event, context):
...
+            data = json.load(json_file)
+            status = data['status']
+            if status == "failed":
+                failed_test += 1
+                tableName = data['labels'][1]['value']
+                failStep = data['steps'][0]['name']
+                description = data['description']
+                open_bug("IPA", tableName[:tableName.find('.')], failStep[:failStep.find('.')], description +
```
Comment (bvolodarskiy) [inline functions/report_push/push_data_report.py:49]: please use String Formatting

File: functions/report_push/jira_events.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    ticketExist = False
+    issues = jira.search_issues('project=' + project_key, maxResults=None)
+    for singleIssue in issues:
+        if summary == str(singleIssue.fields.summary) and str(
+                singleIssue.fields.status) == 'Open':
+            ticketExist = True
+            break
+        if summary == str(singleIssue.fields.summary) and str(
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:23]: elif?

File: functions/report_push/jira_events.py (hunk 2 above)
Comment (oksanagorbachenko) [reply functions/report_push/jira_events.py:15]: done

File: functions/report_push/push_data_report.py (hunk 4 above)
Comment (oksanagorbachenko) [reply functions/report_push/push_data_report.py:41]: done

File: functions/report_push/push_data_report.py (hunk 3 above)
Comment (oksanagorbachenko) [reply functions/report_push/push_data_report.py:39]: done

File: functions/report_push/push_data_report.py (hunk 5 above)
Comment (oksanagorbachenko) [reply functions/report_push/push_data_report.py]: done

File: functions/jira/jira_events.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
+from jira import JIRA
+
+API_URL = os.getenv("JIRA_URL")
+secret_name = os.getenv("SECRET_NAME", "ipa-develop-api-keys")
```
[the remark this answers is not in the document]
Comment (oksanagorbachenko) [reply functions/jira/jira_events.py]: changed to constant

File: functions/report_push/jira_events.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            ticketExist = True
+            print("Will be reopen bug with name [{0}]".format(summary))
+            # jira.transition_issue(singleIssue.key, transition='19')
+            break
+    if not ticketExist:
+        create_new_bug(description, replaced_allure_links, summary)
+
+
+def get_all_issues(jira_project_key):
+    issues = jira.search_issues('project=' + jira_project_key, maxResults=None)
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:35]: str format

File: functions/report_push/jira_events.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    return issues
+
+
+def create_new_bug(description, replaced_allure_links, summary):
+    print("Will be created bug with name [{0}]".format(summary))
+    # jira.create_issue(
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:41]: uncomment

File: functions/report_push/jira_events.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+def open_bug(table_name: str, fail_step: str, description: str, replaced_allure_links, issues):
+    summary = '[DataQA][BUG][{0}]{1}'.format(table_name, fail_step)
+    ticketExist = False
+    for singleIssue in issues:
+        if summary == str(singleIssue.fields.summary) and str(
+                singleIssue.fields.status) == 'Open':
+            ticketExist = True
+            break
+        elif summary == str(singleIssue.fields.summary) and str(
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:24]: i think if actions the same we don't need to elif here

File: functions/report_push/push_data_report.py [hunk 11]
```diff
@@ -5,10 +5,14 @@
 import boto
 import boto.s3
 import re
+import fnmatch
 from datetime import date
 import json
 import awswrangler as wr
 import random
+
+from jira_events import *
```
Comment (bvolodarskiy) [inline functions/report_push/push_data_report.py:14]: import only necessary methods

File: functions/report_push/requirements.txt [hunk 12]
```diff
@@ -9,4 +9,5 @@ fastparquet
 awslambdaric
 awswrangler
 aiobotocore==1.3.0
-# awscliv2==1.5.2
\ No newline at end of file
+# awscliv2==1.5.2
+jira
```
Comment (bvolodarskiy) [inline functions/report_push/requirements.txt:13]: freeze version

File: functions/report_push/jira_events.py [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+                singleIssue.fields.status) != 'Open':
+            ticketExist = True
+            print(f'Will be reopen bug with name[{summary}]')
+            # jira.transition_issue(singleIssue.key, transition='19')
```
Comment (bvolodarskiy) [inline functions/report_push/jira_events.py:28]: uncomment

**Pull request closed without merging**